### ISSUE TITLE
How to test the trained vae model?

### ISSUE THREAD

**Issue by LT1st** [GuHuangAI/DDM-Public#7]
I have trained a version of vae using the `train_vae.py `. And the config is modified from `sketchcoco_ae_kl_256x256_d4.yaml`.
 However, I have no idea how to infer by using vae and verify if my training process is ok.

I am currently use your model for depth estimation. And the sample image in training process makes me consult. I am using rgb to infer depth, but the model predicts rgb instead.
![sample-5](https://github.com/GuHuangAI/DDM-Public/assets/39019137/39febb62-bb41-4ffa-b105-424d35ee6843)

During training the vae model, the loss is pretty high.
```
[Train Step] 24000/50000: train/disc_loss: 0.97690, train/logits_real: 0.04821, train/logits_fake: -0.01064, lr: 0.00000,                                              23800/50000: train/total_loss: 27733.81250, train/logv
[Train Step] 24000/50000: train/disc_loss: 0.97690, train/logits_real: 0.04821, train/logits_fake: -0.01064, lr: 0.00000, :  48%|▍| 24200/50000 [5:39:40<6[Train Step] 0.01523, lr: 0.00000, 24200/50000: train/total_loss: 32116.92578, train/logvar: 0.00000, train/kl_loss: 205429.21875, train/nll_loss: 32068.61328, train/rec_loss: 0.16311, train/d_weight: 5000.00000, train/disc_factor: 1.00000, train/g_loss: 0.00962, lr: 0.00000,                                                                                             24000/50000: train/total_loss: 26278.60156, train/logv
[Train Step] 24200/50000: train/disc_loss: 0.89188, train/logits_real: 0.24713, train/logits_fake: 0.01070, lr: 0.00000,                                               .01807, lr: 0.00000, 
[Train Step] 24200/50000: train/disc_loss: 0.89188, train/logits_real: 0.24713, train/logits_fake: 0.01070, lr: 0.00000, :  49%|▍| 24400/50000 [5:42:29<5:[Train Step] 24400/50000: train/total_loss: 25214.12109, train/logvar: 0.00000, train/kl_loss: 182055.59375, train/nll_loss: 25491.31055, train/rec_loss: 0.12966, train/d_weight: 524200/50000: train/total_loss: 32116.92578, train/logv000.00000, train/disc_factor: 1.00000, train/g_loss: -0.05547, lr: 0.00000,                                                                                            .00962, lr: 0.00000, 
[Train Step] 24400/50000: train/disc_loss: 0.96721, train/logits_real: 0.12701, train/logits_fake: 0.04678, lr: 0.00000, 
[Train Step] 24400/50000: train/disc_loss: 0.96721, train/logits_real: 0.12701, train/logits_fake: 0.04678, lr: 0.00000, :  49%|▍| 24600/50000 [5:45:16<5:[Train Step] 24400/50000: train/total_loss: 25214.12109, train/logv24600/50000: train/total_loss: 23507.39844, train/logvar: 0.00000, train/kl_loss: 201950.67188, train/nll_loss: 24703.16797, train/rec_loss: 0.12565, train/d_weight: 50.05547, lr: 0.00000, 000.00000, train/disc_factor: 1.00000, train/g_loss: -0.23919, lr: 0.00000, 
[Train Step] 24600/50000: train/disc_loss: 0.96768, train/logits_real: 0.31095, train/logits_fake: 0.23294, lr: 0.00000,                                               24600/50000: train/total_loss: 23507.39844, train/logv
[Train Step] 24600/50000: train/disc_loss: 0.96768, train/logits_real: 0.31095, train/logits_fake: 0.23294, lr: 0.00000, :  50%|▍| 24800/50000 [5:48:03<5:[Train Step] 0.23919, lr: 0.00000, 24800/50000: train/total_loss: 22864.04688, train/logvar: 0.00000, train/kl_loss: 215523.43750, train/nll_loss: 22694.06641, train/rec_loss: 0.11543, train/d_weight: 5000.00000, train/disc_factor: 1.00000, train/g_loss: 0.03395, lr: 0.00000,                                                                                             24800/50000: train/total_loss: 22864.04688, train/logv
[Train Step] 24800/50000: train/disc_loss: 0.94819, train/logits_real: 0.10411, train/logits_fake: -0.01646, lr: 0.00000,                                              .03395, lr: 0.00000, 
[Train Step] 24800/50000: train/disc_loss: 0.94819, train/logits_real: 0.10411, train/logits_fake: -0.01646, lr: 0.00000, :  50%|▌| 25000/50000 [5:50:53<9[Train Step] 25000/50000: train/total_loss: 27076.83008, train/logvar: 0.00000, train/kl_loss: 232208.26562, train/nll_loss: 25900.06641, train/rec_loss: 0.13173, train/d_weight: 5000.00000, train/disc_factor: 1.00000, train/g_loss: 0.23531, lr: 0.00000, 
```
I am not sure if my training for vae is right.

**Comment by GuHuangAI** [comment]
There are two stages for training the depth estimation.
1. The vae is trained for the reconstruction of the ground truth. Therefore, you should use depth maps to train the vae rather than RGB images.  The sampled images in the training vae process are the reconstructed results. Like $x^{'} = VAE(x)$.
2. In the second stage, you can follow the ReadMe to train your diffusion models for image to depth.

**Comment by LT1st** [comment]
OK, I am changing my training data. Thanks a lot.

By the way, where can I get the inference scripts for VAE?

**Comment by GuHuangAI** [comment]
You can call the function [validate_img](https://github.com/GuHuangAI/DDM-Public/blob/main/ddm/encoder_decoder.py#L1013) of vae.

**Comment by LT1st** [comment]
I will try it! THANKS
